### PR TITLE
Newsletters: add 217 (2022-09-14)

### PR DESCRIPTION
**Pull request by harding**
- [ ] On Friday (or maybe even Saturday), re-check the mailing lists for any actual news this week @harding 
- [ ] Add lede, releases & RCs, topic links @harding 
- [x] Bitcoin Core PR Review Club @LarryRuane (cc @glozow )
- [x] Bitcoin Core 25614 @adamjonas

### CODE REVIEW

File: _posts/en/newsletters/2022-09-14-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  require that channel open transactions only include *confirmed
+  inputs*---inputs which spend outputs that are a part of a confirmed
+  transaction.  If enabled, this can prevent an initiator from delaying
+  a channel open by basing it off of a large unconfirmed transaction
+  with a low feerate.
+
+- [Eclair #2190][] removes support for the original fixed-length onion
+  data format, which is also proposed for removal from the LN
+  specification in [BOLTs #962][].  The upgraded variable-length format
+  was [added to the specification][bolts #619] more than elmost three
```
Comment (Xekyo) [inline _posts/en/newsletters/2022-09-14-newsletter.md:74]: Which one? "more than three years ago" or "almost three years ago"? :)

**Review by Xekyo — COMMENTED** [review]
Looks good so far, except a merge conflict on that sentence. ;)

**Comment by LarryRuane** [comment]
I still don't have permission to push commits to this branch, but my PR review club changes are here: https://github.com/LarryRuane/bitcoinops.github.io/commit/be1dfa25e9bca03c109473a2d7450ecd1f408b84
(I didn't make a PR, but I can if that would help.)

**Comment by bitschmidty** [comment]
Cherry picked @LarryRuane's PR Review section commit into this branch

**Review by Xekyo — COMMENTED** [review]
LGTM so far, beside the small fix that I propose

File: _posts/en/newsletters/2022-09-14-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+## Bitcoin Core PR Review Club
+
+*In this monthly section, we summarize a recent [Bitcoin Core PR Review Club][]
+meeting, highlighting some of the important questions and answers.  Click on a
+question below to see a summary of the answer from the meeting.*
+
+[Reduce bandwidth during initial headers sync when a block is found][review club 25720]
+is a PR by Suhas Daftuar that reduces a node's network bandwidth
+requirement while synchronizing the blockchain with peers, including
```
Comment (bitschmidty) [inline _posts/en/newsletters/2022-09-14-newsletter.md:24]: Maybe

```suggestion
requirements while synchronizing the blockchain with peers, including
```

File: _posts/en/newsletters/2022-09-14-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+during initial block download (IBD). An important part of the bitcoin
```
Comment (bitschmidty) [inline _posts/en/newsletters/2022-09-14-newsletter.md:25]:
```suggestion
during Initial Block Download (IBD). An important part of the bitcoin
```

Comment (bitschmidty) [inline _posts/en/newsletters/2022-09-14-newsletter.md:25]:
```suggestion
during initial block download (IBD). An important part of the Bitcoin
```

File: _posts/en/newsletters/2022-09-14-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+ethos is minimizing resource demands of running a fully-validating node,
+including networking resources, so as to encourage more users
+to run full nodes. Speeding up the sync time furthers this
+goal as well.
+
+Blockchain synchronization occurs in two phases: First, the node
+receives only block headers from peers; these headers are sufficient
```
Comment (bitschmidty) [inline _posts/en/newsletters/2022-09-14-newsletter.md:32]: Could remove "only" here

File: _posts/en/newsletters/2022-09-14-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+to determine the (likely) best chain (the one with the most work).
+Second, the node uses this best chain to request and download the
```
Comment (bitschmidty) [inline _posts/en/newsletters/2022-09-14-newsletter.md:34]: Maybe 

```suggestion
Second, the node uses this best chain of headers to request and download the
```

File: _posts/en/newsletters/2022-09-14-newsletter.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+corresponding full blocks.
+This PR affects only the first phase (headers download).
+
+{% include functions/details-list.md
+  q0="Why do nodes (mostly) receive `inv` block announcements while
+  they are doing initial headers sync, even though they indicated
+  preference for headers announcements [BIP 130][]?"
```
Comment (bitschmidty) [inline _posts/en/newsletters/2022-09-14-newsletter.md:41]:
```suggestion
  preference for headers announcements ([BIP 130][])?"
```

File: _posts/en/newsletters/2022-09-14-newsletter.md [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+  a1="Each of these peers will then begin sending us the same stream
+  of headers: the `inv` triggers a `getheaders` to the same peer,
+  and its `headers` reply triggers another `getheaders` for the
+  immediately following range of block headers. Receiving duplicate
+  headers is harmless except for the additional bandwidth."
+  a1link="https://bitcoincore.reviews/25720#l-62"
+
+  q2="What would be your estimate (lower/upper bound) of how much
+  bandwidth is wasted?"
+  a2="upper bound (in bytes): `(number_peers - 1) * number_blocks * 81`;
```
Comment (bitschmidty) [inline _posts/en/newsletters/2022-09-14-newsletter.md:58]:
```suggestion
  a2="Upper bound (in bytes): `(number_peers - 1) * number_blocks * 81`;
```

File: _posts/en/newsletters/2022-09-14-newsletter.md [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+  peer is slow, the node times it out (`m_headers_sync_timeout`) and
+  finds another 'initial' headers syncing peer.
+  But if, during headers sync, a node sends an `inv` message
+  that announces blocks, then without this PR, the node will
+  start requesting headers from this peer as well, _without_ setting
+  its `fSyncStarted` flag. This is the source of the redundant
+  headers messages, and was probably not intended, but has the
+  benefit of allowing headers sync to proceed even if the
+  initial headers-sync peer is malicious, broken, or very slow.
+  With the PR, the node requests headers from only _one_ additional
```
Comment (bitschmidty) [inline _posts/en/newsletters/2022-09-14-newsletter.md:82]:
```suggestion
  With this PR, the node requests headers from only _one_ additional
```

**Comment by LarryRuane** [comment]
I adopted @bitschmidty's review suggestions (thanks!) here: https://github.com/LarryRuane/bitcoinops.github.io/commit/4db6e02e79bd1c02f38f8faed8977b3cbddc4783. I created a new commit, rather than rewriting my earlier commit; I hope that's easier to cherry-pick.

**Comment by bitschmidty** [comment]
ACK, squashed. Thanks @harding @LarryRuane @adamjonas @Xekyo ! 🚀

File: _posts/en/newsletters/2022-09-14-newsletter.md [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+## Bitcoin Core PR Review Club
+
+*In this monthly section, we summarize a recent [Bitcoin Core PR Review Club][]
+meeting, highlighting some of the important questions and answers.  Click on a
+question below to see a summary of the answer from the meeting.*
+
+[Reduce bandwidth during initial headers sync when a block is found][review club 25720]
+is a PR by Suhas Daftuar that reduces a node's network bandwidth
+requirements while synchronizing the blockchain with peers, including
+during Initial Block Download (IBD). An important part of the Bitcoin
```
Comment (glozow) [inline _posts/en/newsletters/2022-09-14-newsletter.md:28]: (1) IIUC, the additional peers logic is only for when we're not caught up and not close to tip?
(2) I think "usage" is better than "requirement." The additional peer would only be added if they announced something.
```suggestion
usage
during Initial Block Download (IBD). An important part of the Bitcoin
```

File: _posts/en/newsletters/2022-09-14-newsletter.md [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+
+- [Bitcoin Core #19602][] adds a `migratewallet` RPC that will convert a
+  wallet to natively using [descriptors][topic descriptors].  This works for pre-HD wallets (those
+  created before [BIP32][] existed or was adopted by Bitcoin Core), HD
+  wallets, and watch-only wallets without private keys.  Before calling
+  this function, read the [documentation][managing wallets] and be aware
+  that there are some API differences between non-descriptor wallets and
+  those that natively support descriptors.
+
+<!-- TODO:harding to separate dual funding from interactive funding -->
```
Comment (glozow) [inline _posts/en/newsletters/2022-09-14-newsletter.md:145]: Is this still a todo item?

**Comment by glozow** [comment]
(Sorry for the late review, the content lgtm. The above are just questions.)

File: _posts/en/newsletters/2022-09-14-newsletter.md (hunk 9 above)
Comment (LarryRuane) [reply _posts/en/newsletters/2022-09-14-newsletter.md:28]: > only for when we're not caught up and not close to tip?

Yes, that's correct, and I initially had some text in there to say that, but it seemed too detailed. The main challenge (for me, anyway) in writing these summaries is where to draw the line between precision and concision (readability).

File: _posts/en/newsletters/2022-09-14-newsletter.md [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
+[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
+Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
+Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
+Proposals (BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*
+
+- [Bitcoin Core #25614][] builds on [Bitcoin Core #24464][], allowing
+  the ability to add and trace logs with specific severity levels in
+  addrdb, addrman, banman, i2p, mempool, netbase, net, net_processing,
+  timedata, and torcontrol.
```
Comment (jonatack) [inline _posts/en/newsletters/2022-09-14-newsletter.md:123]: Thanks for covering this change. It added a `-loglevel` configuration option to enable users to select global and per-category severity log levels as part of a larger, ongoing migration to logging with severity levels. It is a "hidden" debug option for now.

**Pull request merged**